mlflow/tracking/_tracking_service/client.py

```diff
@@ -239,17 +240,20 @@ def _record_logged_model(self, run_id, mlflow_model):
                             "{}".format(type(mlflow_model)))
         self.store.record_logged_model(run_id, mlflow_model)
 
+    def _get_artifact_repo(self, run_id):
+        run = self.get_run(run_id)
+        artifact_uri = add_databricks_profile_info_to_artifact_uri(run.info.artifact_uri,
+                                                                   self.tracking_uri)
+        return get_artifact_repository(artifact_uri)
+
     def log_artifact(self, run_id, local_path, artifact_path=None):
         """
         Write a local file or directory to the remote ``artifact_uri``.
 
         :param local_path: Path to the file or directory to write.
         :param artifact_path: If provided, the directory in ``artifact_uri`` to write to.
         """
-        run = self.get_run(run_id)
-        # TODO: use add_databricks_profile_info_to_artifact_uri(artifact_root, self.tracking_uri)
-        #   for DBFS artifact repositories to use the correct tracking URI.
-        artifact_repo = get_artifact_repository(run.info.artifact_uri)
+        artifact_repo = self._get_artifact_repo(run_id)
```

Comment:
Are there client tests that we can add to this to verify the behavior?

Reply:
We could add some unit tests for `_get_artifact_repo` to make sure we call `get_artifact_repository` with the correct URI (once a `dbfs` URI is passed into `get_artifact_repository`, the behavior for returning the correct type of DBFS artifact repository and setting their tracking URIs correctly is pretty well unit-tested). `_get_artifact_repo` is quite trivial currently but since it can change I will add some unit tests.

For artifact URIs that shouldn't be affected by this change, there is some sanity check here for `LocalArtifactRepository`: https://github.com/mlflow/mlflow/blob/55ae14c03d369af10e2a9e975100dfbf86b90d58/tests/tracking/test_rest_tracking.py#L392-L406.